### ISSUE TITLE
DEV - Throws warning when starting desktop

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#40202]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** Dev
**Reproducible in staging?:** dev
**Reproducible in production?:** dev
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @shubham1206agra 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1712819917386169

## Action Performed:
1. Open desktop app in dev environment

## Expected Result:
Opens without any issues

## Actual Result:
Error message is thrown

Onyx DevTools - Error:  ReferenceError: window is not defined
[Electron]     at DevTools.connectViaExtension (webpack://new.expensify/./node_modules/react-native-onyx/dist/DevTools.js?:14:35)
[Electron]     at new DevTools (webpack://new.expensify/./node_modules/react-native-onyx/dist/DevTools.js?:6:31)
[Electron]     at eval (webpack://new.expensify/./node_modules/react-native-onyx/dist/DevTools.js?:69:22)
[Electron]     at ./node_modules/react-native-onyx/dist/DevTools.js (/Users/prachiagrawal/code/App/desktop/dist/main.js:2717:1)
[Electron]     at __webpack_require__ (/Users/prachiagrawal/code/App/desktop/dist/main.js:5411:42)
[Electron]     at eval (webpack://new.expensify/./node_modules/react-native-onyx/dist/Onyx.js?:37:36)
[Electron]     at ./node_modules/react-native-onyx/dist/Onyx.js (/Users/prachiagrawal/code/App/desktop/dist/main.js:2739:1)
[Electron]     at __webpack_require__ (/Users/prachiagrawal/code/App/desktop/dist/main.js:5411:42)
[Electron]     at eval (webpack://new.expensify/./node_modules/react-native-onyx/dist/index.js?:7:32)
[Electron]     at ./node_modules/react-native-onyx/dist/index.js (/Users/prachiagrawal/code/App/desktop/dist/main.js:2816:1)
## Workaround:
unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by mallenexpensify** [comment]
Checking on in #engineering-chat

**Comment by mallenexpensify** [comment]
@pac-guerreiro , 👀 plz, this might be related to 
- https://github.com/Expensify/react-native-onyx/pull/438

**Comment by pac-guerreiro** [comment]
@mallenexpensify Yeah... this is one me 😅 I can take a look into it if you want!

**Comment by mallenexpensify** [comment]
Thanks for the quick reply, assigned to you.  cc @marcaaron and @AndrewGable since you were the reviewers on[ the PR](https://github.com/Expensify/react-native-onyx/pull/438).  Not assigning to either of ya, do so if you want.

**Comment by pac-guerreiro** [comment]
@m-natarajan 

I'm unable to reproduce this. Does the app crash to you? Are you running the app through `npm run desktop`?

**Comment by mallenexpensify** [comment]
Threw `retest-weekly` on here too.

**Comment by pac-guerreiro** [comment]
@m-natarajan 

Can you re-test this? 😄

**Comment by mallenexpensify** [comment]
@pac-guerreiro , @m-natarajan is only responsible for creating issues, they don't monitor or (often) respond to pings once issues are created. Let's wait til the `retest-weekly` label triggers a retest then we'll see what next best steps are.

**Comment by pac-guerreiro** [comment]
@mallenexpensify 

Thanks for the heads up! I didn't know about this, sorry 😅

**Comment by pac-guerreiro** [comment]
@shubham1206agra 

I was able to reproduce this and I'll create a PR with a fix for it 😄

**Comment by pac-guerreiro** [comment]
This issue was caused by [this](https://github.com/Expensify/react-native-onyx/pull/505) PR 😅

**Comment by pac-guerreiro** [comment]
My fix got merged and react-native-onyx was bumped to `v2.0.34`. We'll have to wait until the we bump react-native-onyx version in the app

**Comment by mallenexpensify** [comment]
> We'll have to wait until the we bump react-native-onyx version in the app

Thinking we're waiting on the above.

**Comment by pac-guerreiro** [comment]
We'll have to wait until the we bump react-native-onyx version in the app

**Comment by mallenexpensify** [comment]
Bumped to weekly, per the comment above
> We'll have to wait until the we bump react-native-onyx version in the app

**Comment by mvtglobally** [comment]
QA team is not able to confirm if this issue is still reproducible during KI retests as it requires Development Desktop app. This issue will need to be retested internally

**Comment by pac-guerreiro** [comment]
Once this [PR](https://github.com/Expensify/App/pull/42057) is merged, this issue should be resolved!

**Comment by pac-guerreiro** [comment]
@mvtglobally 

I can retest it once the PR I mentioned above gets merged! 😄

**Comment by pac-guerreiro** [comment]
@mvtglobally @mallenexpensify 

PR is now merged and the issue seems to be resolved! 😄

**Comment by mallenexpensify** [comment]
Thanks @pac-guerreiro , gonna close.  Comment/reopen if you disagree or if the issue persists.

**Comment by pac-guerreiro** [comment]
@mallenexpensify

Looks like the PR that fixed this got reverted in https://github.com/Expensify/App/pull/42725 😅 

I'll keep you posted on any developments about this!

Meanwhile, this issue should show up again

**Comment by pac-guerreiro** [comment]
@mallenexpensify 

New PR is open - https://github.com/Expensify/App/pull/42772 😄

**Comment by pac-guerreiro** [comment]
Still waiting on https://github.com/Expensify/App/pull/42772 😄

**Comment by pac-guerreiro** [comment]
Still waiting on https://github.com/Expensify/App/pull/42772 😄

**Comment by pac-guerreiro** [comment]
@mallenexpensify https://github.com/Expensify/App/pull/42772 is finally merged so the issue should be resolved now! 😄

**Comment by mallenexpensify** [comment]
Thx @pac-guerreiro , comment if anyone's due payment, it doesn't appear so.
